source/fe/mapping_cartesian.cc

```diff
@@ -667,6 +667,21 @@ MappingCartesian<dim,spacedim>::transform (
         for (unsigned int d1=0; d1<dim; ++d1)
           for (unsigned int d2=0; d2<dim; ++d2)
             output[i][d1][d2] = input[i][d1][d2] * data.length[d2]
+                                / data.volume_element;
+      return;
+    }
```

Comment:
Here, only mapping_piola was implemented and the rest of the cases threw a ExcNotImplemented(). Also, mapping_piola was actually doing a mapping_piola_gradient transformation. This does break test cases that used to work (namely, those that use fe_poly_tensor with mapping_piola), but I think it is the right behaviour. Please check.